crates/rust-lib/src/lib.rs

```diff
@@ -572,8 +573,11 @@ pub trait RustGenerator<'a> {
         record: &Record,
         docs: &Docs,
         derive_component: bool,
+        additional_derives: Vec<String>,
     ) {
         let info = self.info(id);
+        // We use a hash set to make sure we don't have any duplicates
+        let additional_derives: HashSet<String> = additional_derives.into_iter().collect();
```

Comment:
One important caveat to be aware of about `HashSet<T>` is that its iteration order is not defined and not deterministic between runs of a process. This means, for example, that different code is generated from the same WIT source as derives might be shuffled around.

Mind using a `BTreeSet` (or `IndexSet`) here instead to preserve order?

Reply:
Oh right, I should have though about the ordering issue. I'll switch that around

Reply:
This should be all resolved, but leaving the comment unresolved until you have a chance to take a look at it